rust/agama-lib/share/profile.schema.json

```diff
@@ -224,54 +219,314 @@
+            "target": {
+              "title": "Target device",
+              "oneOf": [
+                {
+                  "enum": ["disk", "newLvmVg"]
+                },
+                {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": { "disk": { "type": "string" } },
+                  "required": ["disk"]
+                },
+                {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": {
+                    "newLvmVg": {
+                      "type": "array",
+                      "items": { "type": "string" }
+                    }
+                  },
+                  "required": ["newLvmVg"]
+                }
+              ]
+            },
+            "boot": {
+              "title": "Configuration of the boot settings",
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "configure": {
+                  "title": "Whether to configure partitions for booting",
+                  "type": "boolean"
+                },
+                "device": {
+                  "title": "Device to use for booting",
+                  "type": "string"
+                }
+              }
+            },
+            "encryption": {
+              "title": "Encryption settings",
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["password"],
+              "properties": {
+                "password": {
+                  "title": "Passphrase to use when creating new encryption devices",
+                  "type": "string"
+                },
+                "method": {
+                  "title": "Method used to create the encryption devices",
+                  "enum": ["luks2", "tpm_fde"]
+                },
+                "pbkdFunction": {
+                  "title": "Password-based key derivation function to use for LUKS2",
+                  "enum": ["pbkdf2", "argon2i", "argon2id"]
+                }
+              }
+            },
+            "space": {
+              "title": "Policy to find space for the new partitions",
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "policy": {
+                  "enum": ["delete", "resize", "keep", "custom"]
+                },
+                "actions": {
+                  "type": "array"
+                }
+              },
+              "if": {
+                "properties": {
+                  "policy": { "const": "custom" }
+                }
+              },
+              "then": {
+                "required": ["policy", "actions"],
+                "properties": {
+                  "actions": {
+                    "type": "array",
+                    "items": {
+                      "anyOf": [
```

Comment:
In this schema we're using both `anyOf` and `oneOf`, with `or` and `xor` semantics respectively, as I've just learned from https://json-schema.org/understanding-json-schema/reference/combining
AFAICT either of these would work for any place in our schema, so we should be consistent.

Reply:
I think using `anyOf` vs `oneOf` has really meaning if the object is an array. And you are right, in our case we can replace all `oneOf` by `anyOf`. Let's be consistent. Thanks!